compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/bug-aliased-capture-aliased-mutate.expect.md

```diff
@@ -0,0 +1,107 @@
+
+## Input
+
+```javascript
+// @flow @enableTransitivelyFreezeFunctionExpressions:false
+import {arrayPush, setPropertyByKey, Stringify} from 'shared-runtime';
+
+/**
+ * 1. `InferMutableRanges` derives the mutable range of identifiers and their
+ *     aliases from `LoadLocal`, `PropertyLoad`, etc
+ *   - After this pass, y's mutable range only extends to `arrayPush(x, y)`
+ *   - We avoid assigning mutable ranges to loads after y's mutable range, as
+ *     these are working with an immutable value. As a result, `LoadLocal y` and
+ *     `PropertyLoad y` do not get mutable ranges
+ * 2. `InferReactiveScopeVariables` extends mutable ranges and creates scopes,
+ *    as according to the 'co-mutation' of different values
+ *   - Here, we infer that
+ *     - `arrayPush(y, x)` might alias `x` and `y` to each other
+ *     - `setPropertyKey(x, ...)` may mutate both `x` and `y`
+ *   - This pass correctly extends the mutable range of `y`
+ *   - Since we didn't run `InferMutableRange` logic again, the LoadLocal /
+ *     PropertyLoads still don't have a mutable range
+ *
+ * Note that the this bug is an edge case. Compiler output is only invalid for:
+ *  - function expressions with
+ *    `enableTransitivelyFreezeFunctionExpressions:false`
+ *  - functions that throw and get retried without clearing the memocache
+ *
+ * Found differences in evaluator results
+ * Non-forget (expected):
+ *   (kind: ok)
+ *   <div>{"cb":{"kind":"Function","result":10},"shouldInvokeFns":true}</div>
+ *   <div>{"cb":{"kind":"Function","result":11},"shouldInvokeFns":true}</div>
+ * Forget:
+ *   (kind: ok)
+ *   <div>{"cb":{"kind":"Function","result":10},"shouldInvokeFns":true}</div>
+ *   <div>{"cb":{"kind":"Function","result":10},"shouldInvokeFns":true}</div>
+ */
+function useFoo({a, b}: {a: number, b: number}) {
+  const x = [];
+  const y = {value: a};
+
+  arrayPush(x, y); // x and y co-mutate
+  const y_alias = y;
+  const cb = () => y_alias.value;
```

Comment:
Note that this intermediate StoreLocal was not necessary for bug repro before merging hir-rewrite. In #31345, this was a [bug](https://react-compiler-playground-git-pr31345-fbopensource.vercel.app/#N4Igzg9grgTgxgUxALhAejQAgAIIHYCGARgDYIAqMBeYAlgC60BuCJAngGIwIIBeCHKHjiMIeAKIAPAA7cwdMWGQAzAiTAIAOnm3KhI2mMxQNHCBAAU3AgZYBKTMG2ZMcRfUx4IAd0wBeTBIIODUAOR8LOwBuZ2MNAAlzAGtImJ08FzcaD1kEJkMTABECegR-TDAoInoqEWK2MCsEG0Z7NJdYjE8fFzVaAg0AE0x6CExc-OgwYtLYrLAPOQBGcpDpelgEAEk8UYsJgumShAAabu9oufdMOQAmcsj-AD5x7kmi44A6QYI2dtdrnIAMzlAC2UHoxwsXgumC6oNoAHMABYeCAsGDeGAMMoHKYzBDfX7aWLcDYwDIAbWWZzutIQYCBAF1tABfbQgVlAA) without the intermediate StoreLocal.

Now we check the [mutable range of the base identifier](https://github.com/facebook/react/blob/1460d67c5b9a0d4498b4d22e1a5a6c0ccac85fdd/compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts#L344-L349) when deriving hoistable loads